pkg/vault/test/foundry/fork/BoostedPoolQueryVsActualOperation.t.sol

```diff
@@ -0,0 +1,491 @@
+
+import { IBatchRouter } from "@balancer-labs/v3-interfaces/contracts/vault/IBatchRouter.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import { StablePool } from "@balancer-labs/v3-pool-stable/contracts/StablePool.sol";
+import { StablePoolFactory } from "@balancer-labs/v3-pool-stable/contracts/StablePoolFactory.sol";
+
+import { ERC4626RateProvider } from "../../../contracts/test/ERC4626RateProvider.sol";
+import { BaseVaultTest } from "../utils/BaseVaultTest.sol";
+
+contract BoostedPoolQueryVsActualOperationTest is BaseVaultTest {
+    uint256 private constant BLOCK_NUMBER = 6288761;
+
+    address private constant _donorDai = 0x4d02aF17A29cdA77416A1F60Eae9092BB6d9c026;
+    address private constant _donorUsdc = 0x0F97F07d7473EFB5c846FB2b6c201eC1E316E994;
+
+    IERC4626 internal constant wUSDC = IERC4626(0x8A88124522dbBF1E56352ba3DE1d9F78C143751e);
+    IERC4626 internal constant wDAI = IERC4626(0xDE46e43F46ff74A23a65EBb0580cbe3dFE684a17);
+    IERC20 internal usdcFork;
+    IERC20 internal daiFork;
+
+    StablePool internal boostedPool;
+    StablePoolFactory internal stablePoolFactory;
+
+    // For two-token pools with waDAI/waUSDC, keep track of sorted token order.
+    uint256 internal wDaiIdx;
+    uint256 internal wUsdcIdx;
+
+    uint256 internal constant DAI_FACTOR = 1e18;
+    uint256 internal constant USDC_FACTOR = 1e6;
+    uint256 internal constant BUFFER_INIT_AMOUNT = 100;
+    uint256 internal constant DAI_BUFFER_INIT_AMOUNT = BUFFER_INIT_AMOUNT * DAI_FACTOR;
+    uint256 internal constant USDC_BUFFER_INIT_AMOUNT = BUFFER_INIT_AMOUNT * USDC_FACTOR;
+
+    uint256 internal constant BOOSTED_POOL_INIT_AMOUNT = BUFFER_INIT_AMOUNT * 5;
+    uint256 internal constant DAI_BOOSTED_POOL_INIT_AMOUNT = BOOSTED_POOL_INIT_AMOUNT * DAI_FACTOR;
+    uint256 internal constant USDC_BOOSTED_POOL_INIT_AMOUNT = BOOSTED_POOL_INIT_AMOUNT * USDC_FACTOR;
+
+    function setUp() public override {
+        vm.createSelectFork({ blockNumber: BLOCK_NUMBER, urlOrAlias: "sepolia" });
+
+        BaseVaultTest.setUp();
+
+        _setupTokens();
+        _setupLP();
+        _setupBuffers();
+        _createAndInitializeBoostedPool();
+    }
+
+    function testUsdcInWithinBufferExactIn__Fork__Fuzz(uint256 amountIn) public {
+        amountIn = bound(amountIn, (USDC_BUFFER_INIT_AMOUNT) / 10, USDC_BUFFER_INIT_AMOUNT);
+        IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(usdcFork, amountIn, 0);
+
+        uint256 snapshotId = vm.snapshot();
+        _prankStaticCall();
+        (uint256[] memory queryPathAmountsOut, , ) = batchRouter.querySwapExactIn(paths, bytes(""));
+        vm.revertTo(snapshotId);
+
+        vm.prank(lp);
+        (uint256[] memory actualPathAmountsOut, , ) = batchRouter.swapExactIn(paths, MAX_UINT256, false, bytes(""));
```

Comment:
(here and elsewhere) Now that we're at it, maybe it's also worth comparing the other two return values?

Reply:
Maybe it's an overkill; I'd like to hear your opinion about it.

Reply:
Yes, makes sense! Done!